src/Security/Authentication/AuthenticationFailureHandler.php

```diff
@@ -12,48 +12,28 @@
 
 namespace Contao\CoreBundle\Security\Authentication;
 
-use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\Monolog\ContaoContext;
 use Psr\Log\LoggerInterface;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\Session\Session;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Security\Core\Exception\AccountStatusException;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
-use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Security\Http\Authentication\DefaultAuthenticationFailureHandler;
 use Symfony\Component\Security\Http\HttpUtils;
-use Symfony\Component\Translation\TranslatorInterface;
 
 class AuthenticationFailureHandler extends DefaultAuthenticationFailureHandler
 {
-    /**
-     * @var ScopeMatcher
-     */
-    private $scopeMatcher;
-
-    /**
-     * @var TranslatorInterface
-     */
-    private $translator;
-
-    /**
-     * @param HttpKernelInterface  $httpKernel
-     * @param HttpUtils            $httpUtils
-     * @param ScopeMatcher         $scopeMatcher
-     * @param TranslatorInterface  $translator
-     * @param array                $options
-     * @param LoggerInterface|null $logger
-     */
-    public function __construct(HttpKernelInterface $httpKernel, HttpUtils $httpUtils, ScopeMatcher $scopeMatcher, TranslatorInterface $translator, array $options = [], LoggerInterface $logger = null)
+    public function __construct(HttpKernelInterface $httpKernel, HttpUtils $httpUtils, array $options = [], LoggerInterface $logger = null)
     {
-        parent::__construct($httpKernel, $httpUtils, $options, $logger);
+        $this->defaultOptions['failure_path_parameter'] = 'redirect';
```

Comment:
Why hardcoded instead of configurable?

Reply:
We're hardcoding everything that is Contao specific, otherwise the template would need to be synchronized with the firewall configuration and I think that's highly overkill… It's our own firewall listener after all, so there is no need to have parameters that are not to be configured.

Reply:
Agreed.